wordpress-mobile/WordPress-Android
# Small Reader Improvements: Topics Management

Fixes #13972

This PR makes it possible for users to see the `Topic Picker` screen again after they have added some topics. Previously, this screen would only be shown once, within the `Discover` tab and only if the user hasn't followed any topics yet. With this change, even if the user has already followed some topics, they will still be able to see the `Topic Picker` screen through the `Manage Topics & Sites` settings screen.

Discover | Settings
-------|------
<img width="250" height="500" alt="discover" src="https://user-images.githubusercontent.com/9729923/109326115-6ebc6a00-785f-11eb-8686-6830515f9bf1.mp4"> | <img width="250" height="500" alt="settings" src="https://user-images.githubusercontent.com/9729923/109326516-f86c3780-785f-11eb-8e7e-974382b82906.mp4">

To test:
- Unfollow all topics and navigate to `Reader` -> `DISCOVER` tab.
- Make sure the `DISCOVER` screen is empty. Click on the `GET STARTED` button.
- Verify that the `Topic Picker` screen is shown and everything looks as expected. Nothing changes here. Notice the `Choose your interests` title and the `Select a few to continue` text on the disabled state of the button.
- Select a few topics and click the `Done` button.
- Verify that the `DISCOVER` screen is now showing some content.
- Find the ⚙️ icon on top, within the `Reader` toolbar, and click on it to navigate to the `Manage Topics & Sites` settings screen.
- Verify that at least a few topics are listed under the `FOLLOWED TOPICS` tab list.
- Verify that the `cross` icon has been replaced with the `reader following` icon. Click on the `FAB` button.
- Verify that the `Topic Picker` screen is shown and everything looks familiar. Few things change here. Notice that the `Choose your interests` title is no longer shown. Instead the `Follow topics` toolbar title is shown. Also, notice that the `Select a few to continue` text on the disable state of the button has been now replaced with `Done`.
- Select a few topics and click the `Done` button.
- Verify that you get navigated back to the `Manage Topics & Sites` setting screen and that those newly picked topics are now shown within the `FOLLOWED TOPICS` tab list.
- Remove some topics by clicking the `reader following` icon and verify that this functionality works as expected.
- Go wild... play some more, select a lot of topics, select all the topics, verify everything works as expected, then remove lots of topics, remove all topics, check the `DISCOVER` tab again, make sure everything looks and feel okay, continue...

Note:
After talking to @mbshakti, we decided to add another improvement to the `Topics Picker` screen, where we will show some illustration and copy that there are no recommendations left on an empty state. But, this will be done on a separate PR when it will be design ready.

**Merge Instructions** (:warning:):
1. ~~Wait for @mbshakti to do design review.~~
1. ~~Remove the `Needs Design Review` label.~~
1. Merge this PR.

PR submission checklist:
